store/hazelcast/hazelcast.go

```diff
@@ -32,21 +35,37 @@ const (
 
 // HazelcastStore is a store for Hazelcast
 type HazelcastStore struct {
-	hzMap   HazelcastMapInterface
-	options *lib_store.Options
+	mapProvider HazelcastMapInterfaceProvider
+	options     *lib_store.Options
 }
 
 // NewHazelcast creates a new store to Hazelcast instance(s)
-func NewHazelcast(hzMap HazelcastMapInterface, options ...lib_store.Option) *HazelcastStore {
+func NewHazelcast(hzClient *hazelcast.Client, mapName string, options ...lib_store.Option) *HazelcastStore {
+	return &HazelcastStore{
+		mapProvider: func(ctx context.Context) (HazelcastMapInterface, error) {
+			return hzClient.GetMap(ctx, mapName)
+		},
+		options: lib_store.ApplyOptions(options...),
+	}
+}
+
+// newHazelcast creates a new store with given HazelcastMapInterface for test purpose
+func newHazelcast(hzMap HazelcastMapInterface, options ...lib_store.Option) *HazelcastStore {
```

Comment:
I am not a big fan of specific methods just for test cases when it's not needed.

What about using `hzClient HazelcastMapInterface` in `NewHazelcast()` public method and remove this private one? This way, you will be able to mock the client in order to return the value you want for `GetMap()` method (+ you will need to add it to the interface).

What do you think?

Reply:
I agree with you. It can be tested without special methods created for testing.

We need to have `HazelcastClientInterface` with `GetMap` function to be able to mock Hazelcast client. 
Then, `mapProvider` will return `HazelcastMapInterface` using `HazelcastClientInterface#GetMap` function.

Reply:
I agree with you. It is not a good practice to have test-specific functions when not needed.

I couldn't get your point you mentioned how we can mock hazelcast client.

Reply:
Sorry I picked the wrong interface, you’re right the ClientInterface should have a GetMap() method

Reply:
As we cannot mock Hazelcast `Client` (because its GetMap function returns *hazelcast.Map type), I think we need to have a test-specific function for this case.

Reply:
I will take a look at this later. Thank you!